lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -108,7 +116,20 @@ export abstract class ClientApplication {
      * @returns {Promise.<AuthenticationResult | null>} token response or null. If the return value is null, then no auth redirect was detected.
      */
     async handleRedirectPromise(): Promise<AuthenticationResult | null> {
-        return this.isBrowserEnvironment ? this.handleRedirectResponse() : null;
+        if (this.isBrowserEnvironment) {
+            return this.handleRedirectResponse()
+                .then((result: AuthenticationResult) => {
+                    if (result) {
+                        this.broadcastEvent(BroadcastEvent.HANDLE_REDIRECT_SUCCESS, InteractionType.REDIRECT, result);
```

Comment:
I think we should also broadcast the `ACQUIRE_TOKEN_SUCCESS/FAILURE` and `LOGIN_SUCCESS/FAILURE` events here. The tricky thing is going to be determining whether we are returning a login response or a token response. Maybe if we can figure that out the `HANDLE_REDIRECT_SUCCESS/FAILURE` events become redundant?

Reply:
>  The tricky thing is going to be determining whether we are returning a login response or a token response.

In the Angular sample, we did this by looking at whether or not an account is logged in, and if so, then assume the redirect happened as a result of calling acquireTokenRedirect.

Reply:
This may not work for v2 since we support multiple accounts now. We may need to leave ourselves a temp cache entry or include a flag or something in state

Reply:
Right. Something in the state object will likely be simplest.

Reply:
> Is this mainly for telemetry? The event API doesn't really need to distinguish between login/acquireToken for programmatic reasons.

No, this is because people may subscribe to one event over the other and may care if a login happened or a token acquisition happened. I think it's also wise to have consistency between redirect and popup scenarios where possible